.github/workflows/uml.yml

```diff
@@ -0,0 +1,37 @@
+name: Update the UML Diagrams
+
+on:
+  pull_request:
+    types: [closed]
+    branches: [main]
+  push:
+    branches: [automate-uml]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+
+      - name: Update the UML Diagrams
+        run: |
+          sudo apt-get update && sudo apt-get install -y graphviz
+          make uml
+
+      - name: Configure Git
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "github-actions@github.com"
```

Comment:
This doesn't seem to render well. I am not sure of an alternative. Maybe it can be left blank

Reply:
we can leave it blank